linkerd/app/inbound/src/http/server.rs

```diff
@@ -84,7 +94,34 @@ impl<H> Inbound<H> {
                         .push(http::BoxResponse::layer()),
                 )
                 .push(NewAccessLog::layer())
-                .instrument(|_: &T| debug_span!("http"))
+                .check_new_service::<T, http::Request<_>>()
+                .push(svc::ArcNewService::layer())
+        })
+    }
+
+    pub fn push_http_tcp_server<T, I, HSvc>(self) -> Inbound<svc::ArcNewTcp<T, I>>
```

Comment:
maybe a comment explaining what the difference between this and `push_http_server` is?